doc/arch/adr-009-trailing-slash.md

```diff
@@ -0,0 +1,73 @@
+# Decision record: Trailing slash in HTTP resources
+
+## Context
+
+Currently, HTTP resources have many URLs representing the same thing. This is
+not an ideal scenario for search engine indexing or caching. For example, the
+record `GB` resource can be accessed via:
+
+* `GET /records/GB`
+* `GET /records/GB/`
+* `GET /records/GB.json`
+* `GET /records/GB.json/`
+* `GET /records/GB.csv`
+* `GET /records/GB.csv/`
+
+This ADR proposes a canonical URL and a way to consolidate around it.
+
+## Decision
+
+The canonical path for a resource MUST NOT have a trailing slash. Aliases MUST
+redirect (301) to the canonical path.
+
+For example, the `GB` record has the canonical path:
+
+```
+/records/GB
+```
+
+And the canonical JSON resource:
+
+```
+/records/GB.json
+```
+
+Aliases like `/records/GB/` and `/records/GB.json/` MUST redirect to their
+respective canonical resources.
+
+### Paginated resources
+
+The first page for the records list has the canonical path:
+
+```
+/records?page-index=1&page-size=100
```

Comment:
So this means `/records` would 301 to `/records?page-index=1&page-size=100`? I guess this would close off for example the possibility to change the default page size in future as clients would remember this?

Reply:
What do you mean by "would remember this"?

Reply:
> We could also look at defining link rel="canonical" for example to catch cases where non-supported query parameters are used, we would want to point search engines to a single valid URL. Perhaps this would be a separate ADR or RFC though?

This could be part of this ADR and probably a non-normative note in the specification.

Reply:
`301` implies the client does not have to check back with the origin, v.s. `302` where it will check every time.

Reply:
I am not sure about this for index pages, it is quite nice to have a canonical index page without defaults specified. Do you have any other examples of popular APIs that redirect paginated routes to the first page URL like this?
If not, I would rather get the trailing slash part approved but not the paginated resources part.

Reply:
I don't have a popular API example, I'll try to find one. I don't see the "quite nice" part to be honest but I think you have a fair point on focusing on trailing slashes for now and discuss further about pagination.